source-code/plugins/json/src/settings.ts

```diff
@@ -1,47 +1,45 @@
+// pluginOptions for i18next plugin
+export type PluginSettings = {
+	pathPattern: string | Record<string, string>
+	variableReferencePattern?: string[] //default is ["{{", "}}"]
+	ignore?: string[] // default is []
+}
+
 /**
- * Throws an error if the settings are invalid.
+ * Throws an error if the options are invalid.
  *
  * Not using zod becaue it's not worth the bundle size (2kb vs 14kb).
  */
 export function throwIfInvalidSettings(settings: PluginSettings) {
-	if (settings.pathPattern === undefined) {
-		throw new Error(
-			"The pathPattern setting must be defined and include the {language} placeholder. An example would be './resources/{language}.json'.",
-		)
-	} else if (settings.pathPattern.includes("{language}") === false) {
-		throw new Error(
-			"The pathPattern setting must be defined and include the {language} placeholder. An example would be './resources/{language}.json'.",
-		)
-	} else if (settings.pathPattern.endsWith(".json") === false) {
-		throw new Error(
-			"The pathPattern setting must end with '.json'. An example would be './resources/{language}.json'.",
-		)
+	if (typeof settings.pathPattern === "string") {
+		if (settings.pathPattern.includes("{languageTag}") === false) {
```

Comment:
We should also throw an error if the pathPattern includes `{{languageTag}}` (with double brackets), since double brackets are the default for `variableReferencePattern` and they easily can be mixed up as it happened to me.

Reply:
Agree. cc @inlang/plugins

Reply:
Should double brackets or single brackets be the default for `inlang-plugin/json`? We can still change that. We didn't have a default before.

Reply:
depends on what's most used

Reply:
Let's use the single curly braces. They are the most common among JS, TS, Python and C# developers.

Reply:
✅ set `{variable}` set as default for `json`
✅ added error and test for `json` and `i18next`